inlong-sort/sort-core/src/main/java/org/apache/inlong/sort/parser/impl/FlinkSqlParser.java

```diff
@@ -459,20 +460,40 @@ private void parseFilterFields(FilterStrategy filterStrategy, List<FilterFunctio
     /**
      * Parse field relation
      *
-     * @param fields           The fields defined in node
+     * @param fields The fields defined in node
      * @param fieldRelationMap The field relation map
-     * @param sb               Container for storing sql
+     * @param sb Container for storing sql
      */
     private void parseFieldRelations(List<FieldInfo> fields,
             Map<String, FieldRelation> fieldRelationMap, StringBuilder sb) {
         for (FieldInfo field : fields) {
             FieldRelation fieldRelation = fieldRelationMap.get(field.getName());
-            if (fieldRelation != null) {
-                sb.append("\n    ").append(fieldRelation.getInputField().format())
-                        .append(" AS ").append(field.format()).append(",");
-            } else {
+            if (fieldRelation == null) {
                 String targetType = TableFormatUtils.deriveLogicalType(field.getFormatInfo()).asSummaryString();
                 sb.append("\n    CAST(NULL as ").append(targetType).append(") AS ").append(field.format()).append(",");
+                continue;
+            }
+
+            FunctionParam inputField = fieldRelation.getInputField();
+            if (inputField instanceof FieldInfo) {
+                FieldInfo fieldInfo = (FieldInfo) inputField;
+                FormatInfo formatInfo = fieldInfo.getFormatInfo();
+                FieldInfo outputField = fieldRelation.getOutputField();
+                boolean sameType = formatInfo != null
+                        && outputField != null
+                        && outputField.getFormatInfo() != null
+                        && outputField.getFormatInfo().getTypeInfo().equals(formatInfo.getTypeInfo());
```

Comment:
Is this place a short term solution? Is it appropriate to add a CAST operator?

Reply:
If it is just a cast, then there is no neccessary to add a separate cast operator.